src/components/AdvancedSearchBar/__tests__/AdvancedSearchBar.spec.ts

```diff
@@ -0,0 +1,184 @@
+import '@testing-library/jest-dom';
+import AdvancedSearchBar from '../AdvancedSearchBar.vue';
+import { translate } from '@/mixins';
+import { IconName } from '../../Icon';
+import {
+  RenderOptions,
+  render,
+  fireEvent,
+  waitFor
+} from '@testing-library/vue';
+
+const mixins = [translate];
+
+const initialProps = {
+  searchFunction: (searchTerm: string) => {
+    const allResults = [
+      {
+        title: 'Recipients',
+        icon: IconName.USER,
+        results: [
+          {
+            name: 'John Doe',
+            description: 'A postcard to John Doe',
+            type: 'postcard'
+          },
+          {
+            name: 'Jane Doe',
+            description: 'A postcard to Jane Doe',
+            type: 'postcard'
+          },
+          {
+            name: 'John Smith',
+            description: 'soccer postcard',
+            type: 'postcard'
+          }
+        ]
+      },
+      {
+        title: 'Campaigns',
+        icon: IconName.BULLHORN,
+        results: [
+          {
+            name: 'Marketing Campaign for Texas',
+            description: '5000 recipients'
+          },
+          {
+            name: 'Marketing Campaign for California',
+            description: '10000 recipients'
+          }
+        ]
+      },
+      {
+        title: 'Templates',
+        icon: IconName.CREATIVE,
+        results: [
+          {
+            name: 'Template with John Doe',
+            description: 'A template to create postcard for John Doe'
+          },
+          {
+            name: 'Template with Jane Doe',
+            description: 'A template to create postcard for Jane Doe'
+          }
+        ]
+      }
+    ];
+
+    const results = allResults.map((result) => {
+      return {
+        title: result.title,
+        icon: result.icon,
+        results: result.results.filter(
+          (eachResult) =>
+            eachResult.description.includes(searchTerm) ||
+            eachResult.name.includes(searchTerm)
+        )
+      };
+    });
+
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(results);
+      }, 0);
+    });
+  }
+};
+
+// Helper function to render the component
+const renderComponent = (options: RenderOptions) =>
+  render(AdvancedSearchBar, { ...options, global: { mixins } });
+
+describe('AdvancedSearchBar', () => {
+  it('clears entered searchTerm when x button is clicked', async () => {
+    const searchTerm = 'something';
+    const props = {
+      ...initialProps
+    };
+
+    const { queryByTestId, container } = renderComponent({ props });
+
+    const input = container.querySelector('#searchBar');
+    await fireEvent.update(input, searchTerm);
```

Comment:
This is a better test of user input instead of firing the event of the input itself. I won't comment in all the spots but try to utilize the `userEvent` ones over `fireEvent` if possible
```suggestion
    const input = container.querySelector('#searchBar');
    userEvent.type(input, searchTerm);
```